File: JAVA/QuickFind.java
public class QuickFind{
    private int id[];

    QuickFind(int N){
        this.id = new int[N];
        for (int i = 0; i<N; i++){
            this.id[i]=i;
        }
    }

    Boolean connected(int p, int q){
        return this.id[p] == this.id[q];
    }

    void union(int p, int q){
        int pid = this.id[p];
        int qid = this.id[q];

        for (int i = 0; i < this.id.length; i++){
            if (this.id[i] == pid){
                this.id[i] = qid;
            }
        }
    }


    public static void main (String args[]){
        QuickFind qf = new QuickFind(10);
        System.out.println(qf.connected(1,2));
        qf.union(1,2);
        System.out.println(qf.connected(1,2));
    }
}

File: TYPESCRIPT/quickfind.ts
class QuickFindUF {
    private id: number[];
    constructor(N: number) {
        this.id = Array.from({ length: N }, (_, i) => i)
    }

    connected(p: number, q: number): boolean {
        return this.id[p] === this.id[q];
    }

    union(p: number, q: number): void {
        const pid: number = this.id[p];
        const qid: number = this.id[q]

        for (let i = 0; i < this.id.length; i++) {
            if (this.id[i] === pid) {
                this.id[i] = qid;
            }
        }
    }
}

const qf = new QuickFindUF(10);
console.log(qf.connected(1, 2))
qf.union(1, 2)
console.log(qf.connected(1, 2))